reyna/src/com/b2msolutions/reyna/Dispatcher.java

```diff
@@ -244,9 +248,44 @@ private static boolean shouldGzip(Header[] headers) {
         return filteredHeaders.toArray(returnedHeaders);
     }
 
-    private static AbstractHttpEntity getCompressedEntity(String content, Context context) throws Exception {
-        byte[] data = content.getBytes();
-        return AndroidHttpClient.getCompressedEntity(data, context.getContentResolver());
+    private static AbstractHttpEntity getEntity(Message message, Context context) throws Exception {
+        String content = message.getBody();
+
+        byte[] data = content.getBytes("UTF-8");
+        if (data.length <= (AndroidHttpClient.getMinGzipSize(context.getContentResolver()) * 2)){
```

Comment:
Does this mean that we'll gzip everything that is over the gzip minimum regardless of whether they have the gzip header or not?